tests/behat/features/bootstrap/FileContext.php

```diff
@@ -63,4 +74,29 @@ private function getUrlWithGuzzle(array $cookies, string $url) : ResponseInterfa
     return (new Client(['cookies' => $jar]))->get($url);
   }
 
+  /**
+   * Allows getting content from downloaded file.
+   *
+   * @param string $link_text
+   *   The url of file.
+   *
+   * @return string
+   *   The content of the downloaded file.
+   */
+  private function getDownloadedFileContent(string $link_text): string {
+    $session = $this->getSession();
+    $link = $session->getPage()->find('named', ['link', $link_text]);
+    $url = $link->getAttribute('href');
+
+    $hostname = (isset($_SERVER['HTTPS']) ? 'https' : 'http') . '://' . $_SERVER['HTTP_HOST'];
+    if (strpos($url, $hostname) === FALSE) {
+      $url = $hostname . $url;
+    }
+
+    $cookies = $this->getSession()->getDriver()->getCookies();
+    $response = $this->getUrlWithGuzzle($cookies, $url);
+
+    return trim($response->getBody()->getContents());
+  }
```

Comment:
In case we can't re-use the `matches` step but need the new `contains` step, then I suggest two changes to this function:

- Make the method `protected` since we know in Cablecar we can extend the contexts for SaaS specific test steps
- Don't `trim` inside `getDownloadedFileContent` but do it in the calling function. This helps if we ever need to test anything related to downloaded whitespace (like a YAML file).

Reply:
Added